src/ArrayDefinition.php

```diff
@@ -66,10 +66,14 @@ public function withReferenceContainer(?ContainerInterface $referenceContainer):
      */
     public static function fromConfig(array $config): self
     {
+        /**
+         * @psalm-suppress MixedArgument
+         */
         return new self(
             $config[self::CLASS_NAME],
             $config[self::CONSTRUCTOR] ?? [],
-            self::getMethodsAndPropertiesFromConfig($config)
+            // Already normalized config. Ignore all the rest parameters
+            $config['methodsAndProperties'] ?? self::getMethodsAndPropertiesFromConfig($config),
```

Comment:
In this case use `fromPreparedData()` instead of `fromConfig()`.

Reply:
The flow comes from https://github.com/yiisoft/definitions/blob/6060238370d5637076be0a9c44b2e5a624c8c4f3/src/Helpers/Normalizer.php#L86 so it would be weird to unpack there $config there like
```php
return isset($definition['methodsAndProperties']) ? ArrayDefinition::fromPreparedData(
    $config[ArrayDefinition::CLASS_NAME],
    $config[ArrayDefinition::CONSTRUCTOR] ?? [],
    $config['methodsAndProperties']
) : ArrayDefinition::fromConfig($config);
```

Reply:
Where normalized array definition can be passed to `Normalizer::normalize()` ?

Reply:
It this case from the inside `LazyDefinition`

Reply:
Where normalized array definition can be passed to `LazyDefinition` ? 

In configuration array that passed to `LazyDefinition`  must use [array definition syntax](https://github.com/yiisoft/definitions#arraydefinition).

Reply:
> Where normalized array definition can be passed to LazyDefinition ?

https://github.com/yiisoft/di/pull/232/files#diff-55c5d44a849e66cfaefd2086077dfb27d261c84a9736a17c9ff35d285739a6eeR651

Reply:
Suggestion:
- https://github.com/yiisoft/definitions/pull/88
- https://github.com/yiisoft/di/pull/355

Reply:
Merged